hilo.rb

```diff
@@ -1,34 +1,113 @@
 #!/usr/bin/env ruby
-puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
-puts "You will have a total of 6 guesses."
-puts ""
 
-random_num = rand(1..100)
+# Game to guess random number
+class HiLoGame
 
-# gives the user 6 guesses
-for i in 0..5
-    guesses_left = 6 - i
-    puts "You have #{guesses_left} guesses left."
-    puts "Please guess a whole number."
+    # Constructor to initialise the class with a maximum number of guesses
+    def initialize(max_guesses)
+	@max_guesses = max_guesses
+	@guesses_left = @max_guesses
+	@already_guessed = Array.new
+	@random_num = get_random_num
+	welcome
+    end
+
+    def get_random_num
+	random_num = rand(1..100)
+    end
+
+    # Welcome message for the game
+    def welcome
+	puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
+	puts "You will have a total of #{@max_guesses} guesses.\nYou cannot guess the same number more than once."
+	puts ""
+    end
```

Comment:
Most of the these methods should be private so they can't be called outside of the class.